invenio_rdm_records/records/systemfields/files.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2023 CERN.
+# Copyright (C) 2020-2021 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""RDM Record Files field.
+
+The field is implemented here to be able to dump specific fields based on record
+access.
+"""
+
+from invenio_records_resources.records.systemfields.files.field import FilesField
+
+
+class RecordFilesField(FilesField):
```

Comment:
Would it make sense to implement this as a search dumper, that just removes the files? We only care about filtering what is indexed, not e.g. what is rendered for the item (since e.g. when the record owner is making a request they should be able to see the files)